src/main/java/com/maddyhome/idea/vim/group/NotificationService.kt

```diff
@@ -87,7 +96,11 @@ class NotificationService(private val project: Project?) {
         notification,
         "set ideajoin",
         IjOptionConstants.ideajoin
-      ) { VimPlugin.getOptionService().setOption(OptionScope.GLOBAL, IjOptionConstants.ideajoin) }
+      ) {
+        (injector.optionGroup.getOption(IjOptionConstants.ideajoin) as? ToggleOption)?.let { option ->
+          injector.optionGroup.setToggleOption(option, OptionScope.GLOBAL)
+        }
+      }
```

Comment:
I've made some changes to make it a little easier to set options - specifically, introduced `getKnownOption`, `GetKnownStringOption` and `GetKnownToggleOption`. These return an `Option`, do not return null, and can be passed straight to the helper functions to set the option value, set/unset toggles, etc. They must be called with a known option name or they will fail/throw. 

It's a little more work than passing a string name in, but not by much. Using a name based API requires an extra parameter to make the validation error messages work for user facing functionality, and most code that sets options wouldn't have an appropriate value available. And since there are very few places that need to set options, compared to reading options, it's not worth adding an extra set of name based extension functions/overloads to fetch the option and then fetch the value of the option. We can let the caller do this, and it can either do validation of the returned option if it wants, or use the `getKnownOption` functions to show intent and fetch a well-known option that it knows is always available, such as the options listed in `OptionConstants`.